proposals/low-level-struct-improvements.md

```diff
@@ -50,248 +45,226 @@ readonly ref struct Span<T>
-    {
-        _field = ref p;
+        // Okay: The RS(int[]) constructor is NOT marked as [RefFieldEscapes] hence it cannot 
+        // capture any arguments by ref. It is implicitly safe-to-escape to calling method by
+        // all of our existing rules
+        return new RS(new int[1]);
     }
-}
 
-ref struct RS2
-{
-    RS1 _field;
-    public RS2(RS1 p)
+    // This method demonstrates that calling CreateRS2 has the same restrictions as RS(ref int) 
+    // constructor. This demonstrates that factory methods are as flexible as constructors 
+    // concerning `ref` field initialization
+    [RefFieldEscapes] 
+    static RS CreateRS3(ref int parameter)
     {
-        // Okay
-        _field = p;
-    }
+        // Okay: This method has [RefFieldEscapes] hence the safe-to-escape of `rs` is the minimim
+        // of the ref-safe-to-escape of the arguments. In this case that is `parameter` which is 
+        // to the calling method
+        RS rs = new CreateRS2(ref parameter);
+        return rs;
 
-    public RS2(ref int i)
-    {
-        // ERROR: The *safe-to-escape* scope of the constructor here is the 
-        // current method scope while the *safe-to-escape* scope of `this` is
-        // outside the current method scope hence this assignment is illegal
-        _field = new RS1(ref i);
-    }
-
-    public RS2(ref int i)  
-        // ERROR: the *safe-to-escape* return of :this the current method scope
-        // but the 'this' parameter has a *safe-to-escape* outside the current
-        // method scope
-        : this(new RS1(ref i))
-    {
+        int local = 42;
 
+        // Error: In this case the ref-safe-to-escape of `local` is to the current method hence 
+        // it is also the safe-to-escape of the method invocation.
+        return new CreateRS2(ref local);
     }
 }
 ```
 
-The limiting of the constructor rules to just `ref struct` that directly contain `ref` field is another important compatibility concern. Consider that the majority of `ref struct` defined today indirectly contain `Span<T>` references.  That mean by extension they will indirectly contain `ref` fields once `Span<T>` adopts `ref` fields. Hence it's important to ensure the *safe-to-return* rules of constructors on these types do not change. That is why the restrictions must only apply to types that directly contain a `ref` field.
+The samples here have the same patterns as the [compat considerations](#compat-considerations) above. This means it will allow the introduction of `ref` fields without breaking existing code.
 
-Example of where this comes into play.
+Constructor chaining needs to consider these new rules as well. When the original constructor calls a chained constructor via `:this(...)` the chained constructor effectively escapes from the original. That means a chained constructor call is olny legal if the *safe-to-escape* value is not smaller than the original constructor one. This will be accomplished with the following rules:
```

Comment:
```suggestion
Constructor chaining needs to consider these new rules as well. When the original constructor calls a chained constructor via `:this(...)` the chained constructor effectively escapes from the original. That means a chained constructor call is only legal if the *safe-to-escape* value is not smaller than the original constructor one. This will be accomplished with the following rules:
```